[BUG] MultiAgentExample: CustomMergeStrategy example fails to run

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior
The example of a [custom merge strategy for running multiple agents](https://java2ai.com/docs/frameworks/agent-framework/advanced/multi-agent#%E8%87%AA%E5%AE%9A%E4%B9%89%E5%90%88%E5%B9%B6%E7%AD%96%E7%95%A5), where `mergeOutputKey` is not set, causing subsequent serialization to fail.


<img width="1229" height="709" alt="Image" src="https://github.com/user-attachments/assets/ab9b023c-53fd-4363-a33e-ecffdee7d058" />

### Expected Behavior

_No response_

### Steps To Reproduce

1. configure ChatModel
2. run com.alibaba.cloud.ai.examples.documentation.framework.advanced.MultiAgentExample#example4_customMergeStrategy

### Environment

```markdown
Spring AI Alibaba version(s):
```

### Debug logs

执行示例时出错: com.fasterxml.jackson.databind.JsonMappingException: Null key for a Map not allowed in JSON (use a converting NullKeySerializer?)
reactor.core.Exceptions$ReactiveException: com.fasterxml.jackson.databind.JsonMappingException: Null key for a Map not allowed in JSON (use a converting NullKeySerializer?)
	at reactor.core.Exceptions.propagate(Exceptions.java:410)
	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:102)
	at reactor.core.publisher.Mono.block(Mono.java:1779)
	at com.alibaba.cloud.ai.graph.CompiledGraph.invoke(CompiledGraph.java:538)
	at com.alibaba.cloud.ai.graph.agent.Agent.doInvoke(Agent.java:254)
	at com.alibaba.cloud.ai.graph.agent.Agent.invoke(Agent.java:160)
	at com.alibaba.cloud.ai.examples.documentation.framework.advanced.MultiAgentExample.example4_customMergeStrategy(MultiAgentExample.java:288)
	at com.alibaba.cloud.ai.examples.documentation.framework.advanced.MultiAgentExample.runAllExamples(MultiAgentExample.java:489)
	at com.alibaba.cloud.ai.examples.documentation.framework.advanced.MultiAgentExample.main(MultiAgentExample.java:79)
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:104)
		... 7 more
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Null key for a Map not allowed in JSON (use a converting NullKeySerializer?)
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:282)
	at com.fasterxml.jackson.databind.SerializerProvider.mappingException(SerializerProvider.java:1417)
	at com.fasterxml.jackson.databind.SerializerProvider.reportMappingProblem(SerializerProvider.java:1315)
	at com.fasterxml.jackson.databind.ser.impl.FailingSerializer.serialize(FailingSerializer.java:31)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeTypedFields(MapSerializer.java:1024)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeFields(MapSerializer.java:778)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeWithoutTypeInfo(MapSerializer.java:763)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:719)
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:34)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4859)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:4079)
	at com.alibaba.cloud.ai.graph.serializer.plain_text.jackson.JacksonStateSerializer.writeData(JacksonStateSerializer.java:91)
	at com.alibaba.cloud.ai.graph.serializer.StateSerializer.write(StateSerializer.java:55)
	at com.alibaba.cloud.ai.graph.serializer.StateSerializer.write(StateSerializer.java:31)
	at com.alibaba.cloud.ai.graph.serializer.Serializer.objectToBytes(Serializer.java:42)
	at com.alibaba.cloud.ai.graph.serializer.Serializer.writeObject(Serializer.java:61)
	at com.alibaba.cloud.ai.graph.serializer.Serializer.cloneObject(Serializer.java:71)
	at com.alibaba.cloud.ai.graph.serializer.StateSerializer.cloneObject(StateSerializer.java:50)
	at com.alibaba.cloud.ai.graph.CompiledGraph.cloneState(CompiledGraph.java:388)
	at com.alibaba.cloud.ai.graph.GraphRunnerContext.cloneState(GraphRunnerContext.java:296)
	at com.alibaba.cloud.ai.graph.GraphRunnerContext.addCheckpoint(GraphRunnerContext.java:226)
	at com.alibaba.cloud.ai.graph.GraphRunnerContext.buildNodeOutputAndAddCheckpoint(GraphRunnerContext.java:444)
	at com.alibaba.cloud.ai.graph.executor.NodeExecutor.handleActionResult(NodeExecutor.java:169)
	at com.alibaba.cloud.ai.graph.executor.NodeExecutor.lambda$executeNode$1(NodeExecutor.java:118)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onNext(MonoFlatMapMany.java:163)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:121)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:67)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:950)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2340)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:144)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.request(MonoCompletionStage.java:145)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onSubscribe(MonoFlatMapMany.java:141)
	at reactor.core.publisher.MonoCompletionStage.subscribe(MonoCompletionStage.java:56)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:68)
	at reactor.core.publisher.FluxDefer.subscribe(FluxDefer.java:54)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8891)
	at reactor.core.publisher.FluxConcatArray$ConcatArraySubscriber.onComplete(FluxConcatArray.java:238)
	at reactor.core.publisher.FluxConcatArray.subscribe(FluxConcatArray.java:79)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8891)
	at reactor.core.publisher.FluxConcatArray$ConcatArraySubscriber.onComplete(FluxConcatArray.java:238)
	at reactor.core.publisher.FluxConcatArray.subscribe(FluxConcatArray.java:79)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8891)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onNext(MonoFlatMapMany.java:196)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:121)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:67)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:950)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2340)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:144)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.request(MonoCompletionStage.java:145)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onSubscribe(MonoFlatMapMany.java:141)
	at reactor.core.publisher.MonoCompletionStage.subscribe(MonoCompletionStage.java:56)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:68)
	at reactor.core.publisher.FluxDefer.subscribe(FluxDefer.java:54)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8891)
	at reactor.core.publisher.FluxConcatArray$ConcatArraySubscriber.onComplete(FluxConcatArray.java:238)
	at reactor.core.publisher.FluxConcatArray.subscribe(FluxConcatArray.java:79)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8891)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onNext(MonoFlatMapMany.java:196)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:121)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:67)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:950)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2340)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:144)
	at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.request(MonoCompletionStage.java:145)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onSubscribe(MonoFlatMapMany.java:141)
	at reactor.core.publisher.MonoCompletionStage.subscribe(MonoCompletionStage.java:56)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:68)
	at reactor.core.publisher.FluxDefer.subscribe(FluxDefer.java:54)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8891)
	at reactor.core.publisher.FluxConcatArray$ConcatArraySubscriber.onComplete(FluxConcatArray.java:238)
	at reactor.core.publisher.FluxConcatArray.subscribe(FluxConcatArray.java:79)
	at reactor.core.publisher.FluxDefer.subscribe(FluxDefer.java:54)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4576)
	at reactor.core.publisher.Mono.block(Mono.java:1778)
	... 6 more

### Anything else?

_No response_

--- a/examples/documentation/src/main/java/com/alibaba/cloud/ai/examples/documentation/framework/advanced/MultiAgentExample.java
+++ b/examples/documentation/src/main/java/com/alibaba/cloud/ai/examples/documentation/framework/advanced/MultiAgentExample.java
@@ -17,6 +17,7 @@ package com.alibaba.cloud.ai.examples.documentation.framework.advanced;
 
 import com.alibaba.cloud.ai.dashscope.api.DashScopeApi;
 import com.alibaba.cloud.ai.dashscope.chat.DashScopeChatModel;
+import com.alibaba.cloud.ai.graph.GraphResponse;
 import com.alibaba.cloud.ai.graph.OverAllState;
 import com.alibaba.cloud.ai.graph.agent.ReactAgent;
 import com.alibaba.cloud.ai.graph.agent.flow.agent.LlmRoutingAgent;
@@ -245,13 +246,22 @@ public class MultiAgentExample {
 			public Map<String, Object> merge(Map<String, Object> mergedState, OverAllState state) {
 				// 从每个Agent的状态中提取输出
 				state.data().forEach((key, value) -> {
-					if (key.endsWith("_result")) {
+					// 检查key不为null且以"_result"结尾
+					if (key != null && key.endsWith("_result")) {
+						String resultText = "";
+						if (value instanceof GraphResponse graphResponse) {
+                            if (graphResponse.resultValue().isPresent()) {
+                                resultText = graphResponse.resultValue().get().toString();
+                            }
+						} else if (value != null) {
+							resultText = value.toString();
+						}
 						Object existing = mergedState.get("all_results");
 						if (existing == null) {
-							mergedState.put("all_results", value.toString());
+							mergedState.put("all_results", resultText);
 						}
 						else {
-							mergedState.put("all_results", existing + "\n\n---\n\n" + value.toString());
+							mergedState.put("all_results", existing + "\n\n---\n\n" + resultText);
 						}
 					}
 				});
@@ -263,19 +273,19 @@ public class MultiAgentExample {
 		ReactAgent agent1 = ReactAgent.builder()
 				.name("agent1")
 				.model(chatModel)
-				.outputKey("result1")
+				.outputKey("agent1_result")
 				.build();
 
 		ReactAgent agent2 = ReactAgent.builder()
 				.name("agent2")
 				.model(chatModel)
-				.outputKey("result2")
+				.outputKey("agent2_result")
 				.build();
 
 		ReactAgent agent3 = ReactAgent.builder()
 				.name("agent3")
 				.model(chatModel)
-				.outputKey("result3")
+				.outputKey("agent3_result")
 				.build();
 
 		// 使用自定义合并策略
@@ -283,11 +293,16 @@ public class MultiAgentExample {
 				.name("parallel_agent")
 				.subAgents(List.of(agent1, agent2, agent3))
 				.mergeStrategy(new CustomMergeStrategy())
+				.mergeOutputKey("all_results")
 				.build();
 
 		Optional<OverAllState> result = parallelAgent.invoke("分析这个主题");
 
 		if (result.isPresent()) {
+			OverAllState state = result.get();
+			state.value("all_results").ifPresent(mergeResult -> {
+				System.out.println("合并结果: " + mergeResult);
+			});
 			System.out.println("自定义合并策略示例执行成功");
 		}
 	}

